tests/t0124_gateway_ipns_record_test.go

```diff
@@ -0,0 +1,113 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/ipfs/gateway-conformance/tooling/car"
+	. "github.com/ipfs/gateway-conformance/tooling/ipns"
+	"github.com/ipfs/gateway-conformance/tooling/specs"
+	. "github.com/ipfs/gateway-conformance/tooling/test"
+)
+
+func TestGatewayIPNSRecord(t *testing.T) {
+	fixture := car.MustOpenUnixfsCar("t0124/fixtures.car")
+	file := fixture.MustGetRoot()
+	fileCID := file.Cid()
+
+	// test_expect_success "Add the test directory & IPNS records" '
+	// ipfs dag import ../t0124-gateway-ipns-record/fixtures.car &&
+	// ipfs routing put /ipns/${IPNS_KEY} ../t0124-gateway-ipns-record/${IPNS_KEY}.ipns-record
+	// '
+	ipns := MustOpenIPNSRecordWithKey("t0124/k51qzi5uqu5dh71qgwangrt6r0nd4094i88nsady6qgd1dhjcyfsaqmpp143ab.ipns-record")
+	ipnsName := ipns.Key()
+
+	tests := SugarTests{
+		// test_expect_success "Create and Publish IPNS Key" '
+		// curl "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_KEY" > curl_output_filename &&
+		// test_should_contain "Hello IPFS" curl_output_filename
+		// '
+		{
+			Name: "GET an IPNS record from the gateway",
+			Request: Request().
+				Path("ipns/%s", ipnsName),
+			Response: Expect().
+				Body(file.RawData()),
+		},
+		// test_expect_success "GET KEY with format=ipns-record and validate key" '
+		// curl "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_KEY?format=ipns-record" > curl_output_filename &&
+		// ipfs name inspect --verify $IPNS_KEY < curl_output_filename > verify_output &&
+		// test_should_contain "$FILE_CID" verify_output
+		// '
+		// test_expect_success "GET KEY with format=ipns-record has expected HTTP headers" '
+		// curl -sD - "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_KEY?format=ipns-record" > curl_output_filename 2>&1 &&
+		// test_should_contain "Content-Disposition: attachment;" curl_output_filename &&
+		// test_should_contain "Content-Type: application/vnd.ipfs.ipns-record" curl_output_filename &&
+		// test_should_contain "Cache-Control: public, max-age=3155760000" curl_output_filename
+		// '
+		{
+			Name: "GET KEY with format=ipns-record has expected HTTP headers and valid key",
+			Request: Request().
+				Path("ipns/%s", ipnsName).
+				Query("format", "ipns-record"),
+			Response: Expect().
+				Headers(
+					Header("Content-Disposition").Contains("attachment;"),
+					Header("Content-Type").Contains("application/vnd.ipfs.ipns-record"),
+					Header("Cache-Control").Contains("public, max-age=3155760000"),
```

Comment:
@laurentsenta all tests use the same timestamp, humans are lazy, which means some implementers will just hard-code `Cache-Control` and call it a day :P 

Mind adding one more test that has record with slightly different TTL, and call it out in the test Name  that Cache-Control  max-age MUST be based on the TTL from the IPNS Record? (fine to do it in a separate PR)

Reply:
Will do, I created https://github.com/ipfs/gateway-conformance/issues/51